### PR TITLE
Added JWT based authentication

### LINKED ISSUE #603: Add token based authentication in API
A (access) token based auth should be added side-by-side with the current basic auth in the API.
Here are my views on it, 

token - will be a random string stored in the User model as field `token`

To get token the user will send a post request with the credentials to the following urls - 
1. The current url for UI login is `\admin\login` . The view function responsible for it can store the token as cookie if username-password is correct. (as is already done in PR #600 by @niranjan94 )
2. To get token using API, we can create a url like `\api\v2\login` where user details and token will be returned in case of right username-password. 

Any request can then use the token to access protected methods by setting the header `Authorization` 
as `Token <token>`. 

Once a token is created for a user, it will be returned as-is in all login requests and not be re-created. (This will allow a user to be login'd from android and serverUI at the same time). 
Only on 'forgot password' will the the access token be re-created. 

@niranjan94 @shivamMg @rafalkowalski views ?

### PR DESCRIPTION
**Pull request by aviaryan**
Fixes #603, #610 and includes fix for #641

Expiry time of token is set to 1 day.
Auth url is `api/v2/login`
Send there a POST request with email and password to get access_token . 
```
{
  "email": "email@gmail.com",
  "password": "test"
}
```
```
{
  "access_token": "string"
}
```

Include the token in Authorization header as "JWT _token_" for accessing authorized methods.

Note - I have not removed the basic auth. Both work. This is becasue many good services like GitHub allow their APIs to be accessed by both, access token and username:password authentication. Let's follow their example.

### CODE REVIEW

**Comment by aviaryan** [comment]
@shivamMg @niranjan94 @rafalkowalski please review

**Comment by niranjan94** [comment]
Seems to be good. Can be merged. :smile: ... @rafalkowalski @shivamMg What do you think ?

File: open_event/api/helpers.py [hunk 1]
```diff
@@ -170,31 +171,69 @@ def requires_auth(f):
...
     return decorated
+
+
+@jwt_required()
+def auth_jwt():
+    """
+    A helper function that throws JWTError if JWT is not set
+    """
+    g.user = current_identity
+    pass
```
Comment (shivamMg) [inline open_event/api/helpers.py]: No need to pass

File: open_event/__init__.py [hunk 2]
```diff
@@ -51,6 +55,12 @@ def create_app():
     app.logger.setLevel(logging.INFO)
     # logging.getLogger('sqlalchemy.engine').setLevel(logging.INFO)
 
+    # set up jwt
+    app.config['JWT_AUTH_USERNAME_KEY'] = 'email'
+    app.config['JWT_EXPIRATION_DELTA'] = timedelta(seconds=84400)
```
Comment (shivamMg) [inline open_event/__init__.py]: Shouldn't it be 86400?

Comment (aviaryan) [inline open_event/__init__.py]: LOL. Mental maths went wrong.

**Comment by aviaryan** [comment]
@shivamMg @rafalkowalski  updated. please merge if satisfied.

**Pull request merged**